DAOS-13673 object: debug patch for local TX corruption

Nasf-Fan opened the pull request:
Only for test.

Skip-unit-tests: true
Skip-func-test-vm: true

Test-tag-hw-medium-ucx-provider: test_daos_io
Test-repeat-hw-medium-ucx-provider: 10
Skip-func-hw-test-medium-ucx-provider: false

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12686/1/display/redirect

daosbuild1 commented:
Test stage Unit Test bdev with memcheck on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12686/1/display/redirect

daosbuild1 commented:
Test stage Unit Test bdev on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12686/1/display/redirect

daosbuild1 commented:
Test stage Unit Test on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12686/1/display/redirect

daosbuild1 commented:
Test stage NLT on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12686/1/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12686/2/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium UCX Provider completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12686/4/execution/node/855/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12686/5/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12686/6/execution/node/298/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12686/6/execution/node/301/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12686/6/execution/node/332/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12686/7/execution/node/351/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12686/7/execution/node/357/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-12686/7/execution/node/333/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

Nasf-Fan committed:
DAOS-13673 object: debug patch for local TX corruption

Only for test.

Skip-unit-tests: true
Skip-func-test-vm: true

Test-tag-hw-medium-ucx-provider: test_daos_io
Test-repeat-hw-medium-ucx-provider: 10
Skip-func-hw-test-medium-ucx-provider: false

Required-githooks: true

Signed-off-by: Fan Yong <fan.yong@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Functional Hardware Medium Verbs Provider completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-12686/13/testReport/

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.
